JumpscaleLibs/clients/minio/MinioClient.py

```diff
@@ -0,0 +1,114 @@
+"""
+Minio Healing Client
+"""
+
+from Jumpscale import j
+import requests
+
+JSConfigClient = j.baseclasses.object_config
+
+
+class MinioClient(JSConfigClient):
+    """
+    Minio healing client object
+    """
+
+    _SCHEMATEXT = """
+        @url = jumpscale.minio.client
+        name** = "" (S)
+        url** = "" (S)
+        """
+
+    def get_jobs(self):
+        resp = requests.get(f"{self.url}/jobs")
+        resp.raise_for_status()
+        jobs = resp.json()
+    
+        minio_jobs = []
+        for job_id, job in jobs.items():
+            minio_job = MinioJob(
+                self.url, job_id, job["location"], job["type"], job["running"]
+            )
+            minio_jobs.append(minio_job)
+        return minio_jobs
+
+    def heal(self, background=False, file_path=""):
+        url = f"{self.url}/repair"
+        if background:
+            url += "?bg=1"
+            resp = requests.post(url)
+            resp.raise_for_status()
+            return resp.text
+        else:
+            resp = requests.post(url)
+            resp.raise_for_status()
+            return self._read_response_chuncked(resp, file_path)
+    
+    def repair_bucket(self, bucket_name, background=False, file_path=""):
+        url = f"{self.url}/repair/{bucket_name}"
+        if background:
+            url += "?bg=1"
+            resp = requests.post(url)
+            resp.raise_for_status()
+            return resp.text
+        else:
+            resp = requests.post(url)
+            resp.raise_for_status()
+            return self._read_response_chuncked(resp, file_path)
+
+    def repair_bucket_object(self, object_path, background=False, file_path=""):
+        url = f"{self.url}/repair/{object_path}"
+        if background:
+            url += "?bg=1"
+            resp = requests.post(url)
+            resp.raise_for_status()
+            return resp.text
+        else:
+            resp = requests.post(url)
+            resp.raise_for_status()
+            return self._read_response_chuncked(resp, file_path)
+
+    def _read_response_chuncked(self, response, file_path):
+        if file_path == "":
+            return Exception("you must provide a file path")
+        
+        with open(file_path, 'w+') as f:
+            count = 1
+            block_size = 512
+            try:
+                total_size = int(response.headers.get('content-length'))
+            except TypeError:
+                total_size = 10000000
+
+            response.encoding = 'utf-8'
+            for chunk in response.iter_content(chunk_size=block_size, decode_unicode=True):
+                if chunk:
+                    f.write(chunk)                                                                                                                                                                                                                                   
+                    f.flush()
+
+class MinioJob(object):
```

Comment:
This job info is not complete, in case of background job, there are more data. I think it's also better i f u send the entire object from the json to the constructor, and then the job can copy the different parts

> Note: I would just store the job data as private attribute and build `getters` around it to access job info directly from the dict 

something like
```python
@property
def running(self):
   return self.__job['running']
```